src/DataTable/index.jsx

```diff
@@ -52,6 +52,7 @@ function DataTable({
   isLoading,
   children,
   onSelectedRowsChanged,
+  maxSelectableRows,
```

Comment:
[suggestion] I might recommend renaming the prop to `maxSelectedRows` to more clearly depict that this prop controls how many rows may be selected simultaneously, as opposed to limiting the number of selectable rows (e.g., a page of 10 rows with a `maxSelectableRows` of 2 _could_ imply to consumers that even with no user selections made yet, only 2 rows of the 10 would be able to be selected with the remaining 8 being disabled. `maxSelectedRows` (or even `maxUserSelectedRows`) implies the user has made the selections already and the limit applies to the current set of selected rows.